### ISSUE TITLE
Fancy mode breaks on inline images inside wrapped paragraphs.

### ISSUE THREAD

**Issue by jmduke** [buttondown/roadmap#3247]
Consider the following:

```
<p class="foo">
  <img src="..." />
</p>
```

This currently will not render in Fancy Mode, and saving it will destroy the image. Why? 

Because we've explicitly chosen to not allow inline images. 

"Okay," you say, "but why does the class matter?"

Part of the rendering pipeline for Fancy Mode is running the content through `marked`, and one of the things we explicitly compile are isolated images within an empty p tag:

```
      // tiptap doesn’t recognize images wrapped in <p>, so we remove the <p> wrappers around <img>.
      .replace(/<p><img (.+?)><\/p>/g, "<img $1 />")
```

There are a number of paths here:

1. Allow inline images and just have them be read-only.
2. Warn loudly before loading Fancy Mode for these cases
3. Other stuff.

**Comment by rknightuk** [comment]
@jmduke Hey Justin did you get a chance to look at this at all?

**Comment by jmduke** [comment]
@rknightuk Gah I promised you that it would be fixed by Saturday and the nit was not, in fact, fixed by Saturday. Deepest apologies: I will, as penance, fix it this evening and shoot you a reproduction.

**Comment by jmduke** [comment]
Okay, a good path forward will be bumping our version of tiptap to 2.5 so we can take advantage of [richer error handling](https://github.com/ueberdosis/tiptap/pull/5178), and then I think just loudly fail with a prompt to drop down to Markdown. 

Robb, this won't be fixed by tonight just because we need to test the minor version bump but if you remove the `app-icons` class in the short term you should be able to avoid the issue entirely.

**Comment by rknightuk** [comment]
Is there anything we can do temporarily that isn't removing the class - because we're using RSS import that class is used to style stuff on the web version?

**Comment by jmduke** [comment]
Yup — just don't toggle into Fancy Mode. As long as you look at the email in Markdown, you'll be fine.

**Comment by rknightuk** [comment]
Ah okay, we'll do that then 👍
